src/insidersUpdate/updateCheck.ts

```diff
@@ -0,0 +1,301 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ */
+
+import * as nls from 'vscode-nls';
+const localize: nls.LocalizeFunc = nls.loadMessageBundle();
+
+import * as vscode from "vscode";
+import * as https from "https";
+import * as fs from "fs";
+import * as semver from "semver";
+import { HttpsProxyAgent } from "https-proxy-agent";
+import { OutgoingHttpHeaders, ClientRequest } from 'http';
+import * as URL from 'url';
+import { Utilities } from "../utilities";
+import { GitHubRelease, GitHubApiBase, GitHubAsset, GitHubRateLimitResponse } from "./gitHubInterfaces";
+
+/**
+ * The flow of this update checking is as follows.
+ * Use the GitHub APIs to git a list of releases for the repository.
+ * https://developer.github.com/v3/repos/releases/#list-releases-for-a-repository
+ * Then look at the "tag_name" in the repositories which are created in the form
+ * "v<semver>-insiders" for insiders builds.
+ * Compare the semantic version of the release candidate to the extensions current version.
+ * If candidate version is greater than the extension version, then find the correct assets
+ * for that release and find the VSIX.
+ * This is done using this GitHub API.
+ * https://developer.github.com/v3/repos/releases/#list-assets-for-a-release
+ * Once we find the asset with the correct "name", it is downloaded and installed.
+ * This implementation was adapted from the Microsoft C/C++ extension.
+ * https://github.com/microsoft/vscode-cpptools/blob/58c50dc38b1a3ebcae8139b28c0904d468d11e6e/Extension/src/githubAPI.ts
+ */
+
+const gitHubRepo: string = 'Microsoft/sarif-vscode-extension';
+const vsixAssetName: string = 'Microsoft.Sarif-Viewer.vsix';
```

Comment:
Broader than this review, but we should align the VSIX file name with our extension ID `ms-sarifvscode.sarif-viewer`.

Reply:
Honestly not sure why the ID isn't Microsoft.SarifViewer.. But yeah, we can discuss changing it althought the VSIX file name isn't really relavent.

---
In reply to: [416030790](https://github.com/microsoft/sarif-vscode-extension/pull/267#discussion_r416030790) [](ancestors = 416030790)

Reply:
It's not, but I thought I had the wrong extension for a second.

Reply:
Btw, the "Full" Visual Studio Sarif extension downloads as "Microsoft.Sarif.Viewer.vsix".